central/authprovider/telemetry/telemetry.go

```diff
@@ -23,25 +23,25 @@ var Gather phonehome.GatherFunc = func(ctx context.Context) (map[string]any, err
 			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
 			sac.ResourceScopeKeys(resources.Access)))
 	props := make(map[string]any)
-
-	providers, err := datastore.Singleton().GetAllAuthProviders(ctx)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get AuthProviders")
-	}
-
-	providerIDTypes := make(map[string]string, len(providers))
+	providerIDTypes := make(map[string]string)
 	providerTypes := set.NewSet[string]()
 	providerOriginCount := map[storage.Traits_Origin]int{
 		storage.Traits_DEFAULT:              0,
 		storage.Traits_IMPERATIVE:           0,
 		storage.Traits_DECLARATIVE:          0,
 		storage.Traits_DECLARATIVE_ORPHANED: 0,
 	}
-	for _, provider := range providers {
+
+	err := datastore.Singleton().ForEachAuthProvider(ctx, func(provider *storage.AuthProvider) error {
 		providerIDTypes[provider.GetId()] = provider.GetType()
 		providerTypes.Add(provider.GetType())
 		providerOriginCount[provider.GetTraits().GetOrigin()]++
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get AuthProviders")
```

Comment:
nit (wording): `get` might not be the appropriate term anymore here. Could something like `categorize` or `process ... for telemetry` be more appropriate now ?